### PR TITLE
(PA-1138) Add el-7-ppc64le (Power8) platform

### PR DESCRIPTION
**Pull request by ScottGarman**
Two things worth noting here:

1. There was an existing platform definition file, that was originally a community contribution. It has been absorbed into the official way we build things now.

2. This commit includes multiple rbconfigs for different ruby versions (2.1.9/2.3.3/2.4.1), to facilitate the backport to stable.

### CODE REVIEW

File: configs/components/facter.rb [hunk 1]
```diff
@@ -110,7 +110,7 @@
     skip_blkid = 'OFF'
   elsif platform.is_rpm?
     if (platform.is_el? && platform.os_version.to_i >= 6) || (platform.is_sles? && platform.os_version.to_i >= 11) || platform.is_fedora?
-      pkg.build_requires "libblkid-devel" unless platform.architecture == "s390x"
+      pkg.build_requires "libblkid-devel" unless platform.is_cross_compiled?
```
Comment (Magisus) [inline configs/components/facter.rb]: This won't catch any additional things, will it? Are we not cross-compiling anything else besides s390x and now this new one?

Comment (McdonaldSeanp) [inline configs/components/facter.rb]: We cross compile many things, including solaris, AIX, huawei, cisco....

Comment (ScottGarman) [inline configs/components/facter.rb]: I'll update this with a comment - the reason this is necessary is if we have libblkid-devel installed, facter will try to link against the native (x86_64) library and not the one in our sysroot. So we want to avoid having this package installed in all cross-compiled situations.

**Comment by Magisus** [comment]
Why are there two rbconfigs for 2.1.9? What is the difference between the `gnu` and not `gnu` versions?

**Comment by ScottGarman** [comment]
@Magisus I snuck an additional rbconfig in there for the Ubuntu-based Power8 builds, since we're going to want to enable stable builds for these platforms. I can move that out into a separate PR if that seems clearer.

**Comment by Magisus** [comment]
Ah gotcha. Maybe not a separate PR, but maybe a separate commit describing the fact that it's for a different Power8 OS?

**Comment by McdonaldSeanp** [comment]
Yeah, either a second commit, or a longer description (including justification for the second rbconfig) will do

**Comment by ScottGarman** [comment]
@Magisus good call, I've separated that out into its own commit now.

**Comment by McdonaldSeanp** [comment]
From a vanagon perspective the changes look sound

**Comment by McdonaldSeanp** [comment]
👍

**Comment by Magisus** [comment]
Thanks! 👍

File: configs/components/facter.rb (hunk 1 above)
Comment (Magisus) [inline configs/components/facter.rb]: Wait, I thought we did not cross compile AIX? (this is unrelated to this line of code)

**Comment by jcoconnor** [comment]
@ScottGarman Looks ok based on our recent pairing/intro on same.
